testing/scripts/have-spicy

```diff
@@ -1,6 +1,2 @@
 #!/bin/sh
-if grep -q "DISABLE_SPICY:BOOL=true" "${BUILD}"/CMakeCache.txt; then
-    return 1
-else
-    true
-fi
+"${BUILD}/zeek-config" --have_spicy >/dev/null
```

Comment:
I *think* this needs to be `--have-spicy-analyzers` instead - those tests are skipped for me currently and probably also in CI.
After fixing, the test fails with an ASAN build on my end due to the following: https://github.com/zeek/spicy/issues/1310

I think we `--disable-spicy` for the Zeek ASAN build in CI (not sure why that was, actually), so it's not blocking for this PR. It is generally unfortunate that due to zeek/spicy#1310 spicy-dump, spicy-driver, etc from a Zeek ASAN-enabled are not usable.

Reply:
> I _think_ this needs to be `--have-spicy-analyzers` instead - 

Good catch, I renamed the option. Will update.

> I think we `--disable-spicy` for the Zeek ASAN build in CI (not sure why that was, actually), so it's not blocking for this PR.

Agree, it's an existing problem, something to look into, but I'll leave that for later. FWIW, we do ASAN builds on the Spicy side.